### PR TITLE
Add timestamp logits processor for whisper

### PR DESCRIPTION
**Pull request by stevenlix**
Enable timestamp estimation and logits processing for Whisper model.

### CODE REVIEW

File: docs/ContribOperators.md [hunk 1]
```diff
@@ -451,6 +451,8 @@ This version of the operator has been available since version 1 of the 'com.micr
 <dd>Mask of vocabulary for first step. Words that masked with 0 are not allowed to be generated, and 1 is allowed. Shape is (batch_size, vocab_size)</dd>
 <dt><tt>attention_mask</tt> (optional) : I</dt>
 <dd>Custom attention mask. Shape is (batch_size, sequence_length)</dd>
+<dt><tt>timestamp_enable</tt> (optional) : tensor(bool)</dt>
```
Comment (tianleiwu) [inline docs/ContribOperators.md:454]: How about enable_timestamp

File: onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc [hunk 2]
```diff
@@ -238,6 +238,128 @@ void PresencePenaltyLogitsProcessor<T>::Process(const ISequences*,
...
+    : eos_token_id_(eos_token_id), max_initial_timestamp_index_(max_initial_timestamp_index) {}
+
+template <typename T>
+void TimestampLogitsProcessor<T>::Process(const ISequences* sequences,
+                                          NextTokenScores<T>& next_token_scores) {
+  const int beg_token_id_ = eos_token_id_ + 107;
+  const int not_token_id_ = eos_token_id_ + 106;
+  const int solm_token_id_ = eos_token_id_ + 105;
+  const int sot_token_id_ = eos_token_id_ + 1;
+  constexpr int translate_token_id_ = 50358;
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc:252]: These constant configuration (50358 etc) shall read from attributes

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 3]
```diff
@@ -1095,6 +1095,7 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BeamSearch, 1,
                                 .Input(7, "vocab_mask", "Mask of vocabulary. Words that masked with 0 are not allowed to be generated, and 1 is allowed. Shape is (vacab_size)", "M", OpSchema::Optional)
                                 .Input(8, "prefix_vocab_mask", "Mask of vocabulary for first step. Words that masked with 0 are not allowed to be generated, and 1 is allowed. Shape is (batch_size, vocab_size)", "M", OpSchema::Optional)
                                 .Input(9, "attention_mask", "Custom attention mask. Shape is (batch_size, sequence_length)", "I", OpSchema::Optional)
+                                .Input(10, "timestamp_enable", "Enable timestamp processing. True means enabled. Shape is (1). Default  is ``False``", "tensor(bool)", OpSchema::Optional)
```
Comment (tianleiwu) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:1098]: Maybe we can export  time stamp processing to a subgraph, then call subgraph for logits processing. In this way, user can add its own logits processing

File: onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc (hunk 2 above)
Comment (stevenlix) [reply onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc:252]: eos_token_id_  is from attribute, tokens here have constant offset from eos_token_id_ and may not need to be provided explicitly in the attributes

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 3 above)
Comment (stevenlix) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:1098]: Interesting idea. will think about it.

File: docs/ContribOperators.md (hunk 1 above)
Comment (stevenlix) [reply docs/ContribOperators.md:454]: If we put timestamp_enable in attributes, users will need to regenerate model if they want to switch timestamp on and off.

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 3 above)
Comment (tianleiwu) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:1098]: Another solution is to change the inferface to more general, like logits_processor of tensor(int). So we can use value 1 to indicate time stamp. If other model has its own processing, it could use 2 etc for new processor.

**Review by yufenglee — DISMISSED** [review]
:shipit:

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc (hunk 3 above)
Comment (stevenlix) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: Done

**Review by yufenglee — APPROVED** [review]
